src/hooks/scrollToAnchor.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react'
+
+/**
+ * This scrolls anchor links to their top position with a buffer to
+ * compensate for our sticky nav height.
+ */
+export const useScrollToAnchor = (): void => {
+    useEffect(() => {
+        const scrollToElement = setTimeout(() => {
+            const { hash } = window.location
+
+            if (hash) {
+                const element = document.querySelector(hash)
+                const elementTop = element?.getBoundingClientRect().top
+                const elementPosition: number =
+                    (typeof elementTop === 'number' && elementTop + window.scrollY) || window.scrollY
+
+                const topNav = document.querySelector('.navbar')
+                const topNavHeight: number = topNav?.getBoundingClientRect().height || 0
+                const scrollToPosition: number = elementPosition - topNavHeight
+
+                window.scrollTo({
+                    top: scrollToPosition,
+                    behavior: 'smooth',
+                })
+            }
+        }, 100)
+
+        return () => clearTimeout(scrollToElement)
+    }, [])
+}
```

Comment:
👏🏻 

I'm curious why this is implemented as a custom hook as opposed to a function running on mount in `_app`, could you explain?

Reply:
I didn't want to bloat the app.tsx file, and since we can't necessarily use a collocated structure yet, it felt like a good place to put it for now until Next.js releases their layouts update.